pkg/operator/audit/audit.go

```diff
@@ -0,0 +1,186 @@
+package audit
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"strings"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	kyaml "k8s.io/apimachinery/pkg/util/yaml"
+	auditv1beta1 "k8s.io/apiserver/pkg/apis/audit/v1beta1"
+
+	"github.com/openshift/cluster-kube-apiserver-operator/pkg/operator/v410_00_assets"
+	libgoapiserver "github.com/openshift/library-go/pkg/operator/configobserver/apiserver"
+)
+
+const (
+	defaultConfigAsset = "v4.1.0/config/defaultconfig.yaml"
+	policyAsset        = "v4.1.0/kube-apiserver/audit-policies-cm.yaml"
+)
+
+func NewAuditPolicyPathGetter() (libgoapiserver.AuditPolicyPathGetterFunc, error) {
+	policies, err := readPolicyNamesFromResource()
+	if err != nil {
+		return nil, err
+	}
+
+	return func(profile string) (string, error) {
+		const (
+			auditPolicyFilePath = "/etc/kubernetes/static-pod-resources/configmaps/kube-apiserver-audit-policies"
+		)
+
+		// we expect the keys for audit profile in bindata to have lower case.
+		key := strings.ToLower(profile)
+		_, exists := policies[key]
+		if !exists {
+			return "", fmt.Errorf("invalid audit profile - key=%s", key)
+		}
+
+		return fmt.Sprintf("%s/%s", auditPolicyFilePath, key), nil
+	}, nil
+}
+
+// NewDefaultConfigWithDefaultPolicy injects the default audit policy into the default
+// config for apiserver. This is required for bootstrapping purposes only.
+func NewDefaultConfigWithDefaultPolicy() ([]byte, error) {
```

Comment:
why should audit package know something about a default config? The other way around would make sense IMO.